app/src/main/java/com/framgia/chat_03/screen/chat/ChatActivity.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    private void initPresenter() {
+        UserRepository userRepository = new UserRepository(
+                new UserLocalDataSource(PreferenceManager.getDefaultSharedPreferences(this)),
+                new UserRemoteDataSource(FirebaseAuth.getInstance(),
+                        FirebaseDatabase.getInstance(),
+                        FirebaseStorage.getInstance()));
+        MessageRepository messageRepository = new MessageRepository(new MessageRemoteDataSource(
+                FirebaseAuth.getInstance(), FirebaseDatabase.getInstance()));
+        mPresenter = new ChatPresenter(messageRepository, userRepository);
+        mPresenter.setView(this);
+        mPresenter.loadCurrentUser();
+        mPresenter.loadInteractUser(mInteractUser.getUid());
+        mPresenter.loadMessageFromDataBase(mInteractUser.getUid());
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mPresenter.onStart();
+    }
+
+    @Override
+    protected void onStop() {
+        mPresenter.onStop();
+        super.onStop();
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.image_add:
+                if (isTextMessageZoomOut) {
+                    zoomInTextMessage();
+                }
+                break;
+            case R.id.image_send:
+                break;
+            case R.id.image_photo:
+                break;
+            case R.id.image_camera:
+                break;
+            case R.id.image_back:
+                break;
+        }
+    }
+
+    @Override
+    public void showProgressDialog() {
+        mDialog.setMessage(getString(R.string.login_loading));
+        mDialog.show();
+    }
+
+    @Override
+    public void hideProgressDialog() {
+        mDialog.dismiss();
+    }
+
+    @Override
+    public void onGetMessagesSuccess(List<Message> messages) {
+        mAdapter.addData(messages);
+        if (messages.size() > 0) {
+            mRecyclerView.smoothScrollToPosition(messages.size() - 1);
+        }
+    }
+
+    @Override
+    public void showInformationUser(User user) {
+        try {
+            mTextName.setText(user.getName());
+            Glide.with(this)
+                    .load(user.getImage())
+                    .apply(RequestOptions.circleCropTransform())
+                    .into(mImageAvatar);
+            if (user.isOnline()) {
+                mImageStateUser.setImageResource(R.drawable.bg_dot_online);
+                mTextLastOnline.setText(getResources().getString(R.string.state_online));
+            } else {
+                mImageStateUser.setImageResource(R.drawable.bg_dot_offline);
+                CharSequence timeAgo = DateUtils.getRelativeTimeSpanString(user.getLastOnline(),
+                        System.currentTimeMillis(), DateUtils.MINUTE_IN_MILLIS);
+                mTextLastOnline.setText(timeAgo);
+            }
+        } catch (NullPointerException e) {
```

Comment:
có thể xử lý bằng cách check
if (user ==null) return

Reply:
Em gặp vấn đề là khi em logout và cập nhật trạng thái của currentUser thì tất cả những chỗ em dùng hàm getCurrentUserFromDataBase nó sẽ nhảy vào hàm onData và Glide.with(this) nó báo lỗi context bị null anh ạ

Reply:
e cần xác định vấn đề ở đâu và xử lý
nếu null ở user thì cần xử lý user
còn nếu null context thì xử lý context